### PR TITLE
Added CodeGen for C# (RestSharp)

### LINKED ISSUE #131: Add CodeGen for C# (RestSharp)
**Tell us about the task you want to perform and are unable to do so because the feature is not available**
Add code generator for C#
Library - https://restsharp.dev/

### PR DESCRIPTION
**Pull request by Tilakraj-B**
## PR Description
added CodeGen for C# using RestSharp

## Related Issues

- Related Issue #131 
- Closes #131 

### Checklist
- [X] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [X] I have updated my branch and synced it with project `main` branch before making this PR
- [X] I have run the tests (`flutter test`) and all tests are passing

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests have not been included_

### CODE REVIEW

**Comment by animator** [comment]
Provide all setup & testing instructions to run the generated code on machine locally.

**Comment by Tilakraj-B** [comment]
for setting up c# env in vs code : https://code.visualstudio.com/docs/csharp/get-started

after setting up go to the directory which contains Program.cs and install RestSharp using dotnet add package RestSharp

for testing you can paste the code generated in Program.cs and debug it using F5

File: lib/utils/http_utils.dart [hunk 1]
```diff
@@ -160,3 +160,13 @@ String? formatBody(String? body, MediaType? mediaType) {
   }
   return null;
 }
+
+String getBaseUrl(String url) {
```
Comment (animator) [inline lib/utils/http_utils.dart:164]: Why are you using these utils when we already have `getValidRequestUri()`

File: lib/utils/http_utils.dart [hunk 2]
```diff
@@ -160,3 +160,13 @@ String? formatBody(String? body, MediaType? mediaType) {
...
+  Uri uri = Uri.parse(url);
+  return "${uri.scheme}://${uri.host}${uri.hasPort ? ":${uri.port}" : ""}${uri.hasFragment ? "#${uri.fragment}" : ""}";
+}
+
+String getUrlPath(String url) {
```
Comment (animator) [inline lib/utils/http_utils.dart:169]: Same as above.

**Comment by Tilakraj-B** [comment]
I overlooked the getValidRequestUri() function both the functions can be removed.

**Comment by animator** [comment]
You did not run all the codes locally and test it ..
Why is code generated in POST 1 failing to compile?

**Comment by Tilakraj-B** [comment]
I thought the link in the request model were dummy but now I tested all the requests and solved the error in the post request and updated the tests.

**Comment by animator** [comment]
Thanks for the PR.

**Pull request merged**